sklearn/feature_selection/_sequential.py

```diff
@@ -125,11 +146,24 @@ def __init__(
 
         self.estimator = estimator
         self.n_features_to_select = n_features_to_select
+        self.tol = tol
         self.direction = direction
         self.scoring = scoring
         self.cv = cv
         self.n_jobs = n_jobs
 
+        if self.n_features_to_select == "warn":
+            # for backwards compability
+            warnings.warn(
+                "Leaving n_features_to_select to "
+                "None is deprecated in 1.0 and will become 'auto' "
+                "in 1.2. To keep the same behaviour as with None "
+                "(i.e. select half of the features) and avoid "
+                "this warning, you should manually set "
+                "n_features_to_select='auto' and set tol=None.",
+                PendingDeprecationWarning,
```

Comment:
Usually, we are using a `FutureWarning`. Is there a reason for not using it here?

Reply:
There is no particular reason, so I fixed it with https://github.com/scikit-learn/scikit-learn/pull/20145/commits/0d1b5f3fde10e1f0ee5bb664fbdd740bd7b9f03d